### PR TITLE
docs(codepipeline-actions): clarify how to use an encrypted existing Bucket with the S3SourceAction

### PR DESCRIPTION
**Pull request by berenddeboer**
If you import an S3 bucket encrypted with a custom key, it's impossible to give the source action role permission to use that key for decryption as far as I could see. I.e. I was not able to access this source action role.

By adding `encryptionKey` you can specify it at in the S3 source props, and permissions can then be granted inside this private heavy object. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the PR @berenddeboer, but the way to do that in the CDK is to specify the Key when importing a Bucket:

```ts
const bucket = s3.Bucket.fromBucketAttributes(this, 'Bucket', {
  arn: 'arn:aws:s3:...',
  encryptionKey: kms.Key.fromKeyArn(stack, 'MyKey', 'arn:aws:kms:...'),
});
```

Now, passing `bucket` to `S3SourceAction` will grant the action's Role permissions to use that Key.

(BTW, if you want to customize the permissions a Role for a given Action has, you can always pass in a Role when creating the Action by using the  `role` property of the Action class)

Hope this clears this up!

**Comment by berenddeboer** [comment]
Ah that's great, I completely missed that. Shall I just change my pull request to a documentation update? I.e. giving an example?

And would it be better to submit a new pull request for that?

**Comment by skinny85** [comment]
> Ah that's great, I completely missed that. Shall I just change my pull request to a documentation update? I.e. giving an example?
> 
> And would it be better to submit a new pull request for that?

Feel free to either repurpose this one to only have a documentation update, or close this one, and open a new one - I'm fine with either, whatever is easier for you 🙂.

**Comment by dnz-bdeboer** [comment]
Linter now fails as I removed the test file change. Is it possible to change the title to a docs update? It seems I can't change the title myself.

File: packages/@aws-cdk/aws-codepipeline-actions/README.md [hunk 1]
```diff
@@ -218,6 +218,24 @@ const sourceBucket = s3.Bucket.fromBucketAttributes(this, 'SourceBucket', {
...
+  encryptionKey: key,
+});
+const sourceAction = new codepipeline_actions.S3SourceAction({
+  actionName: 'S3Source',
+  bucket: sourceBucket,
+  bucketKey: 'path/to/file.zip',
+  encryptionKey: encryptionKey,
+  output: sourceOutput,
+});
+```
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/README.md:237]: All of this should be removed.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/s3/source-action.ts [hunk 2]
```diff
@@ -72,7 +72,17 @@ export interface S3SourceActionProps extends codepipeline.CommonAwsActionProps {
...
+   * The Amazon S3 bucket that stores the source code.
+   *
+   * If you import an encrypted bucket in your stack, please specify
+   * the encryption key at import time, for example:
+   *
+   * ```
+   * const bucket = s3.Bucket.fromBucketAttributes(this, 'Bucket', {
+   *   BucketName: 'MyBucket',
+   *    encryptionKey: kms.Key.fromKeyArn(stack, 'MyKey', 'arn:aws:kms:...'),
+   * });
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/s3/source-action.ts:84]: Let's not put code in the docs (remember that CDK is a multi-language environment).

Just mention the `Bucket.fromBucketAttributes()` method here, and that you should provide the Key there.

Or move these docs to the ReadMe file (those are automatically translated to other languages).

**Review by skinny85 — CHANGES_REQUESTED** [review]
> Linter now fails as I removed the test file change. Is it possible to change the title to a docs update? It seems I can't change the title myself.

Done!

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @berenddeboer!

**Pull request merged**